lib/web_ui/lib/initialization.dart

```diff
@@ -141,3 +141,32 @@ class PlatformViewRegistry {
 
 /// The platform view registry for this app.
 final PlatformViewRegistry platformViewRegistry = PlatformViewRegistry();
+
+/// A registry for textures.
+class TextureRegistry {
```

Comment:
We are working on removing web-specific API from `dart:ui` because it confuses compilers and code analyzers. Instead, we expose API through JS-interop. Let's consult with @ditman on how best to do this for textures. We might want to do it in a way that's easy to expose through `flutter.js` API.

Reply:
Realistically we don't have a good place to put this in a way that is ergonomic for users to use (and a workaround for the analyzer to freak out less when using these web-only APIs).

In practice, the API surface of dart:ui has barely changed in the last couple of years, and adding one more object to the mix is not too bad.

@yjbanov I think this can land in `dart:ui`, and whenever we come up with a better location for all these extra methods (`dart:web_ui`?), move them there. WDYT?

Reply:
@skreborn you're not alone, and this is such a big change that you'll know when it's coming for sure! The decision of making the flutter engine (`dart:ui`) a dart library predates me, by years! I guess it's because the engine needs some extra help from the Dart compilers, but I haven't dug too deep.

(I'd be happy if we could get to a point where `dart:ui` does not have anything web-specific, and `dart:web_ui` (or whatever its final name is) has only web-specific things, so it's clear from your imports if you're calling web-only APIs. Not sure what needs to be done for that to happen, haven't looked into the structural bits of the code yet!)

Reply:
**PS: We now have a `dart:ui_web` package for things like this 😅**

Thank @eyebrowsoffire and @mdebbar!

* https://github.com/flutter/engine/tree/main/lib/web_ui/lib/ui_web/src